app/models/project_log.rb

```diff
@@ -37,9 +37,21 @@ class ProjectLog < ActiveRecord::Base
 
   validates :project_step_id, presence: true
 
-  def name
-    # logger.debug "this: #{self.inspect}"
-    "#{project_step.try(:name)} log"
+  def self.filter_by(params)
+    if params[:org].present?
+      # TODO: this will have to be updated when BasicProjects are added
+      joins(project_step: :loan).where(loans: { organization_id: params[:org] })
+    else
+      all
+    end
+  end
+
+  def self.in_division(division)
+    if division
+      joins(project_step: :loan).where(loans: { division_id: division.self_and_descendants })
```

Comment:
No need to load all these descendant divisions into memory. Would `division.self_and_descendants.pluck(:id)` work?